src/Icon/Icon.jsx

```diff
@@ -9,28 +9,51 @@ import animationSpeeds from '../assets/animationSpeeds'
 import { default as DefaultIcon } from '@mdi/react'
 import sizes from '../assets/sizes'
 
-export const Icon = ({ name, size, color, rotate, ...props }) => {
-  const glyph = useMemo(() => {
+let currentGlyphs = { ...glyphs }
+
+export const addIcon = (name, icon) => {
+  if (currentGlyphs[name]) {
+    console.warn(`trying to redefine icon ${name}`)
+    return
```

Comment:
Let the consumer redefine the icon, maybe this is the intended action:

```suggestion
    console.warn(`Warning: existing icon ${name} is overwritten`)
```